apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -7,37 +7,61 @@
    * This is PinAppNavigation's constructor.
    * @param {Object} list [DOM container that contains pin elements]
    */
-  function PinAppNavigation(list) {
-    if (!list && typeof list !== 'object') {
-      console.error('List of pin apps must be an object');
-      return;
+  function PinAppNavigation(list, selector) {
+    if (!list) {
+      console.error('List must be a dom container with navigated elements');
     }
 
-    this.elemList = list;
-    window.removeEventListener('keydown', this);
-    window.addEventListener('keydown', this);
-  }
+    /**
+     * This variable contains all elements that will get using this.selector
+     * @type {Array}
+     */
+    this.elements = [];
 
+    /**
+     * Index of element in this.elements that is selected
+     * @type {Number}
+     */
+    this.selectedElemIndex = 0;
 
-  PinAppNavigation.prototype = {
+    /**
+     * DOM container that has all elements
+     * @type {Dom container}
+     */
+    this.elemList = null;
 
-    selector: '',
-    elements: null,
-    preventDef: false,
-    selectedElemIndex: 0,
-    elemList: null,
-    prevVerticalKey: null,
+    /**
+     * This variable is used to know selected element
+     * has middle size or large size
+     * @type {Boolean}
+     */
+    this.middleElem = null;
+
+    this.elemList = list;
 
-    middleElem: false,
+    /**
+     * This is a selector that used for get all pinned icons on main screen
+     * @type {String}
+     */
+    if (selector && typeof selector === 'string'){
+      this.selector = selector;
+      this.refresh();
+    }else{
+      this.selector = '';
+    }
+    //window.addEventListener('keydown', this);
+  }
 
 
+  PinAppNavigation.prototype = {
+
     /**
-     * This function reset all pinns to default values.
+     * This function resets Main homscreen's state to default state (with clock)
```

Comment:
This function resets Main homscreen's state to default ~~state~~one (with clock)